py_ecc/optimized_bls12_381/optimized_pairing.py

```diff
@@ -161,5 +161,15 @@ def pairing(Q: Optimized_Point3D[FQ2],
     return miller_loop(twist(Q), cast_point_to_fq12(P), final_exponentiate=final_exponentiate)
 
 
-def final_exponentiate(p: Optimized_Field) -> Optimized_Field:
-    return p ** ((field_modulus ** 12 - 1) // curve_order)
+exptable = [FQ12([0] * i + [1] + [0] * (11 - i)) ** field_modulus for i in range(12)]
```

Comment:
Nitpick: how about make it more like a constant
```suggestion
EXPTABLE = tuple([FQ12([0] * i + [1] + [0] * (11 - i)) ** field_modulus for i in range(12)])
```